### DISCUSSION TITLE
[Suggestions] Save the state of the Interface Type in the save file

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#6690] [Ideas]
I suggest to save the "Interface Type" state (currently "Good" and "Evil") in the save file so that the "Interface Type" is automatically adjusted when loading a save file.

In fheroes2:

![Interface type in save file 01](https://user-images.githubusercontent.com/43583994/220604500-89a8a096-dbaf-4534-8ed5-41ef876b0e05.png)

**Comment by oleg-derevenetz** [comment]
This means that if I in general prefer the Evil interface, I should still watch the Good interface when loading the user's save files? Not very convenient. IMHO save files should not store any UI settings at all.

**Reply by LeHerosInconnu** [reply]
@oleg-derevenetz.

> This means that if I in general prefer the Evil interface, I should still watch the Good interface when loading the user's save files? Not very convenient. IMHO save files should not store any UI settings at all.

A player will play mainly his own games that he has already saved.
If the choice of this player is to select a different type of interface, for his own reasons, with different scenarios, I think it would be nice if the player would end up directly with the interface he chose when he was playing this scenario.

**Reply by Branikolog** [reply]
Hi, @LeHerosInconnu 
I personally ok with current implementation. Adding UI state to a savefile looks quite complicated and not really much useful. (In my opinion.)
But if we decide to implement such interface states for savefiles, we can activate them only for dynamic interface (which we're currently missing), So if a player prefers "good" or "evil" interface he can set in options the preferred one. In a case "dynamic" was chosen, the game would change interface type while starting new game or loading a particular save file (with saved UI state).
